### PR TITLE
Feature: Improvements to the scheduling page.

### PR DESCRIPTION
**Pull request by iSOLveIT**
Tasks handled in this PR:

- [x] Added option on the schedule page to show conference schedule in your own timezone.
- [x] Ensured activities are well arranged under their respective days and times.
- [x] Added talk duration field to the submit talk form.
- [x] Integrated the backend into the frontend design.
- [x] Made the activity cards a hyperlink so that when you click on an activity it takes you to a page that shows the full details about an activity
- [x] Added check to ensure both activity start and end times are between 09:00 to 18:00 and to ensure the minute hand for a start or end time is either o'clock or 30 minutes. Eg: A valid value for start will be 11:00 or 11:30

Factors that were considered in implementing this feature:
- The start and end times for a conference day are 09:00 am to 06:00 pm, respectively.
- The duration of an activity is derived from the end and start times of a talk which the user provides when submitting a talk
- Tracks are not implemented so the activities are arranged using the time they fall under and their level.
- Ability to export schedules in the iCal calendar would be added soon.
- Wrote the code in Python so I used the zoneinfo module to handle timezones details but added a backport for Python 3.6+ < 3.9 versions.

Kindly let me know if you encounter any errors.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -35,3 +35,5 @@ Werkzeug==1.0.1
 WTForms==2.3.3
 WTForms-Alchemy==0.17.0
 WTForms-Components==0.10.5
+tzdata
```
Comment (Abdur-rahmaanJ) [inline requirements.txt:38]: please pin the exact version!

File: traveller/modules/cfp/view.py [hunk 2]
```diff
@@ -55,6 +57,12 @@ def add_talk(year):
     talk.year = year
     talk.talk_conference = conf
 
+    start_time, end_time = request.form.get('start_time', "09:00"), request.form.get('end_time', "18:00")
```
Comment (Abdur-rahmaanJ) [inline traveller/modules/cfp/view.py:60]: just a confirmation, if someone wants to add a time before 09:00 can it be done? the 09:00 is just a default value?

**Comment by Abdur-rahmaanJ** [comment]
@iSOLveIT 

The tests fail with this error code, see details above.

`AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'`

If you can rewrite the tests or do something similar so that we don't have to drop py3.6

**Comment by Abdur-rahmaanJ** [comment]
@iSOLveIT though sorting by level is a viable choice, i think adding tracks is the way to go forward. As we must have the flexibility to schedule talks as we want even a beginner talk after an advanced one or vice versa

File: traveller/modules/cfp/view.py (hunk 2 above)
Comment (iSOLveIT) [reply traveller/modules/cfp/view.py:60]: The 09:00 is a just default value but you can't add a time before 9:00. This setting can be changed, though, depending on the time intended time you guys want the conference to start.

File: requirements.txt (hunk 1 above)
Comment (iSOLveIT) [reply requirements.txt:38]: Okay

**Comment by iSOLveIT** [comment]
> @iSOLveIT
> 
> The tests fail with this error code, see details above.
> 
> `AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'`
> 
> If you can rewrite the tests or do something similar so that we don't have to drop py3.6

I altered the test files and conftest file to make things work. This error is just an error with Python 3.6. I research on how to solve it but I found out that the datetime module in Python 3.6 doesn't include the 'fromisoformat' method so I think dropping Python 3.6 will be ok. We can then add Python 3.9 to the test suite.



> @iSOLveIT though sorting by level is a viable choice, i think adding tracks is the way to go forward. As we must have the flexibility to schedule talks as we want even a beginner talk after an advanced one or vice versa

Okay, so with the activities, we do not sort them by level but by the start time of the activities in ascending order. If two or more activities are with the same start time, then we sort the activities using the position of an activity in the sorted list.

Anyways, I am working on how to add tracks so we can use it to schedule talks.

File: traveller/modules/cfp/view.py (hunk 2 above)
Comment (Abdur-rahmaanJ) [reply traveller/modules/cfp/view.py:60]: @iSOLveIT keep it flexible so that we can add times whenever we wish.

Comment (iSOLveIT) [reply traveller/modules/cfp/view.py:60]: Can you explain further? I don't really get what you mean by that statement.  
I wanted to make it flexible, but there is no field in the database which states the start and end times for a conference. Would have used that.

**Pull request closed without merging**